examples/pixels-admin/src/providers/authProvider.ts

```diff
@@ -1,128 +1,225 @@
+                        provider: providerName,
+                    });
+
+                if (error) {
+                    return {
+                        success: false,
+                        error,
+                    };
+                }
+
+                if (data?.url) {
+                    return {
+                        success: true,
+                    };
+                }
             }
 
-            if (data?.url) {
-                return Promise.resolve(false);
-            }
-        }
+            const { data, error } =
+                await supabaseClient.auth.signInWithPassword({
+                    email,
+                    password,
+                });
 
-        const { data, error } = await supabaseClient.auth.signInWithPassword({
-            email,
-            password,
-        });
+            if (error) {
+                return {
+                    success: false,
+                    error,
+                };
+            }
 
-        if (error) {
-            return Promise.reject(error);
-        }
+            const role = data.user?.app_metadata?.role;
 
-        const role = data.user?.app_metadata?.role;
+            if (role === "editor" || role === "admin") {
+                return {
+                    success: true,
+                };
+            }
 
-        if (role === "editor" || role === "admin") {
-            return Promise.resolve();
+            await supabaseClient.auth.signOut();
+
+            return {
+                success: false,
+                error: new Error("User not authorized."),
+            };
+        } catch (error) {
+            return {
+                success: false,
+                error: new Error("Login failed"),
+            };
         }
-
-        await supabaseClient.auth.signOut();
-
-        return Promise.reject(new Error("User not authorized."));
     },
     register: async ({ email, password }) => {
-        const { data, error } = await supabaseClient.auth.signUp({
-            email,
-            password,
-        });
+        try {
+            const { data, error } = await supabaseClient.auth.signUp({
+                email,
+                password,
+            });
 
-        if (error) {
-            return Promise.reject(error);
-        }
+            if (error) {
+                return {
+                    success: false,
+                    error,
+                };
+            }
 
-        const { user } = data;
-        if (user) {
-            return Promise.resolve();
+            const { user } = data;
+            if (user) {
+                return {
+                    success: true,
+                };
+            }
+
+            return {
+                success: false,
+                error: new Error("Register failed"),
+            };
+        } catch (error) {
+            return {
+                success: false,
+                error: new Error("Register failed"),
```

Comment:
```suggestion
                error,
```

It might be better to throw an error from catch. but I do not know. catch can send an hard to read message.